app/Http/Controllers/BookController.php

```diff
@@ -43,20 +44,14 @@ public function store(Request $request)
             'price' => 'required|numeric|min:0',
             'genre' => 'required|string|max:255',
             'subgenre' => 'required|string|max:255',
-            'writer_id' => 'required|exists:writers,id',
-            'publisher_id' => 'required|exists:publishers,id',
+            'writer_id' => 'nullable|exists:writers,id',
+            'writer_name' => 'nullable|string|max:255',
```

Comment:
With both `writer_id` and `writer_name` being `nullable`, the user could try creating a Book without a writer. Please add additional validation that at least one field needs to be present.